src/Gui/GuiApplication.cpp

```diff
@@ -237,57 +237,83 @@ bool GUISingleApplication::isRunning() const
     return d_ptr->running;
 }
 
-bool GUISingleApplication::sendMessage(const QByteArray &message, int timeout)
+bool GUISingleApplication::sendMessage(const QString &message, int timeout)
 {
     QLocalSocket socket;
     bool connected = false;
     for(int i = 0; i < 2; i++) {
         socket.connectToServer(d_ptr->serverName);
         connected = socket.waitForConnected(timeout/2);
-        if (connected || i > 0)
+        if (connected || i > 0) {
             break;
+        }
         int ms = 250;
 #if defined(Q_OS_WIN)
         Sleep(DWORD(ms));
 #else
         usleep(ms*1000);
 #endif
     }
-    if (!connected)
+    if (!connected) {
         return false;
+    }
 
-    QDataStream ds(&socket);
-    ds << message;
-    socket.waitForBytesWritten(timeout);
-    return true;
+    QTextStream ts(&socket);
+#if QT_VERSION <= QT_VERSION_CHECK(6, 0, 0)
+    ts.setCodec("UTF-8");
+#else
+    ts.setEncoding(QStringConverter::Utf8);
+#endif
+#if QT_VERSION <= QT_VERSION_CHECK(5, 15, 0)
+    ts << message << endl;
+#else
+    ts << message << Qt::endl;
+#endif
+
+    bool bytesWritten = socket.waitForBytesWritten(timeout);
+    socket.close();
+    if (socket.state() == QLocalSocket::UnconnectedState ||
+        socket.waitForDisconnected()) {
```

Comment:
Lint warning, the if is empty.

Reply:
Ah yes, thanks, I missed this one when removing debugging code.